src/components/com_tjcertificate/administrator/models/certificate.php

```diff
@@ -188,4 +188,80 @@ public function getCertificateProviderInfo($contentId, $client)
 
 		return trim(implode("\n", $html));
 	}
+
+	/**
+	 * Method to delete record
+	 *
+	 * @param   array  $certificateIds  post data
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function delete($certificateIds)
+	{
+		$certificateIds  = (array) $certificateIds;
+		$table    = $this->getTable('Certificates');
+
+		foreach ($certificateIds as $certificateId)
+		{
+			$table->load(array('id' => (int) $certificateId));
+
+			if ($table->delete($table->id))
+			{
+				if ($table->is_external)
+				{
+					$dispatcher = \JEventDispatcher::getInstance();
+					$dispatcher->trigger('onExternalCertificateAfterDelete', array($table));
+				}
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Publish the element
+	 *
+	 * @param   array  $ids    Item id
+	 * 
+	 * @param   int    $state  Publish state
+	 *
+	 * @return  boolean
+	 * 
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function publish($ids, $state = 1)
+	{
+		$table = $this->getTable();
+
+		foreach ($ids as $id)
+		{
+			$table->load($id);
+			$table->state = $state;
+
+			if ($table->store())
+			{
+				if ($table->is_external)
+				{
+					JLoader::import('components.com_tjcertificate.events.record', JPATH_SITE);
+					$tjCertificateTriggerRecord = new TjCertificateTriggerRecord;
+
+					$tjCertificateTriggerRecord->onRecordStateChange($table, $table->state);
```

Comment:
As discussed, comment this for now...if you work on the states like Pending, Active etc, then you enable it

Reply:
just added trigger and notification already removed